src/Octoshift/VersionChecker.cs

```diff
@@ -23,7 +24,24 @@ public VersionChecker(HttpClient httpClient, OctoLogger log)
         public async Task<bool> IsLatest()
         {
             var curVersion = GetCurrentVersion();
-            var latestVersion = await GetLatestVersion();
+            var latestVersion = string.Empty;
+
+            var cacheCheck = Environment.GetEnvironmentVariable("OCTOSHIFT_VERSION_CHECK_DATE");
+
+            if (DateTime.TryParse(cacheCheck, out var cacheDate))
+            {
+                if (cacheDate.AddDays(1) > DateTime.Now)
+                {
+                    _log.LogVerbose("Version check cache is valid");
+                    latestVersion = Environment.GetEnvironmentVariable("OCTOSHIFT_VERSION_CHECK_RESULT");
+                }
+            }
+            else
+            {
+                latestVersion = await GetLatestVersion();
+                Environment.SetEnvironmentVariable("OCTOSHIFT_VERSION_CHECK_DATE", DateTime.Now.ToString());
```

Comment:
I believe this will only work in Windows. It won't be able to physically create an env variable in macOS and Linux!